app/src/androidTest/java/com/android/sample/ui/endtoend/ActivityFlowTest.kt

```diff
@@ -289,4 +289,77 @@ class ActivityFlowTest {
     hlp.see(ACTIVITY_ROW)
     hlp.notSee(PLUS_BUTTON_TO_CREATE)
   }
+
+  @Test
+  fun aUserLikesAnActivity() {
+    // fun aUserLikesCommentsAndEditsAnActivity() {
+    val email = defaultUserCredentials2["email"]!!
+    val password = defaultUserCredentials2["password"]!!
+
+    // Signs in
+    hlp.write(Auth.SignIn.EMAIL_INPUT, email)
+    hlp.write(Auth.SignIn.PASSWORD_INPUT, password)
+    hlp.click(Auth.SignIn.SIGN_IN_BUTTON)
+
+    // Overview screen
+    hlp.see(Overview.SCREEN)
+
+    // Likes screen
+    hlp.click(BottomNavigation.Liked, bottomNavItem = true)
+    hlp.notSee(Overview.ACTIVITY_CARD)
+    hlp.see("emptyLikedActivityPrompt")
+
+    // Overview screen
+    hlp.click(BottomNavigation.OVERVIEW, bottomNavItem = true)
+    hlp.click(Overview.ACTIVITY_CARD, any = true, index = 1)
+
+    // Activity details screen
+    hlp.see(ActivityDetails.SCREEN)
+    hlp.click("likeButton" + "false")
+    hlp.see("likeButton" + "true")
+    hlp.click(ActivityDetails.GoBackButton)
+
+    // Overview screen
+    hlp.see(Overview.SCREEN)
+
+    // Likes screen
+    hlp.click(BottomNavigation.Liked, bottomNavItem = true)
+    composeTestRule.waitForIdle()
+    hlp.see(Overview.ACTIVITY_CARD)
+    hlp.notSee("emptyLikedActivityPrompt")
+    hlp.click(Overview.ACTIVITY_CARD)
+
+    // Activity details screen
+    // Amine Dafer: for your PR, you will need to uncomment the following lines
+    /*hlp.click(ActivityDetails.COMMENT_SECTION) // COMMENT_SECTION is not defined in the TestTags.kt file, you will need to add it. It is the tag of the comment section in the ActivityDetails screen
+    hlp.notSee(ActivityDetails.COMMENT_ITEM)
+    hlp.write(ActivityDetails.COMMENT_INPUT, "This is a comment")
+    hlp.click(ActivityDetails.COMMENT_POST_BUTTON)
+    hlp.see("This is a comment", text = true)
+    hlp.see(ActivityDetails.COMMENT_ITEM)
+    hlp.click(ActivityDetails.EDIT_BUTTON)*/
+
+    // Edit activity screen
+    // Should just fix tags and scroll behavior
+    /*hlp.see(EditActivity.SCREEN)
+    hlp.write(EditActivity.TITLE_INPUT, "Edited Activity Title", replace = true)
+    hlp.write(EditActivity.DESCRIPTION_INPUT, "Edited Activity Description", replace = true)
+    hlp.write(EditActivity.PRICE_INPUT, "15", replace = true)
+
+    val location = e2e_locations["Geneva"]!!.shortName
+    hlp.scroll(EditActivity.SCREEN, EditActivity.LOCATION_INPUT)
+    hlp.write(EditActivity.LOCATION_INPUT, location, replace = true)
+    hlp.see(EditActivity.LOCATION_ITEM, any = true)
+    hlp.click(location)
+    hlp.scroll(EditActivity.SCREEN, EditActivity.EDIT_BUTTON)
+    hlp.click(EditActivity.EDIT_BUTTON)
+
+    // Activity details screen
+    hlp.see(Overview.SCREEN)
+    hlp.see("Edited Activity Title", text = true)
+    hlp.click("Edited Activity Title", text = true)
+    hlp.see("Edited Activity Description", text = true)
+    hlp.see("15.0 CHF", text = true)
+    hlp.see(location, text = true)*/
+  }
```

Comment:
It is unfortunate we couldn't add these! But I guess that is fine, the end to end already looks good enough! And there are other good e2e too.